pkg/controller/warehouses/warehouses.go

```diff
@@ -304,6 +382,13 @@ func (r *reconciler) syncWarehouse(
 				Reason:  "AwaitingFreightCreation",
 				Message: "Freight creation from latest artifacts is in progress",
 			},
+			&metav1.Condition{
+				Type:               kargoapi.ConditionTypeCriteriaSatisfied,
+				Status:             metav1.ConditionTrue,
+				Reason:             "CriteriaMet",
+				Message:            "freight creation criteria satisfied",
+				ObservedGeneration: warehouse.GetGeneration(),
+			},
 		)
 
 		// Build a Freight from the latest discovered artifacts.
```

Comment:
Not something you did, but while we're here... "a Freight" isn't grammatically correct.

Alternatively, "a Freight resource" would be acceptable as well.

```suggestion
		// Build Freight from the latest discovered artifacts.
```